Source/Urho3D/Math/Ray.cpp

```diff
@@ -251,17 +251,20 @@ float Ray::HitDistance(const void* vertexData, unsigned vertexStride, unsigned v
     unsigned index = 0, nearestIdx = M_MAX_UNSIGNED;
     Vector3 barycentric;
     Vector3* outBary = outUV ? &barycentric : nullptr;
+    Vector3 tmpOutNormal, *tmpOutNormalPointer = outNormal ? &tmpOutNormal : nullptr;
 
     while (index + 2 < vertexCount)
     {
         const Vector3& v0 = *((const Vector3*)(&vertices[index * vertexStride]));
         const Vector3& v1 = *((const Vector3*)(&vertices[(index + 1) * vertexStride]));
         const Vector3& v2 = *((const Vector3*)(&vertices[(index + 2) * vertexStride]));
-        float distance = HitDistance(v0, v1, v2, outNormal, outBary);
+        float distance = HitDistance(v0, v1, v2, tmpOutNormalPointer, outBary);
         if (distance < nearest)
         {
             nearestIdx = index;
             nearest = distance;
+            if (outNormal)
+                *outNormal = tmpOutNormal;

```

Comment:
`nearest` is defined as `M_INFINITY` so won't `outNormal` always get defined if it exists (no matter what `distance` is for the first iteration, the conditional always evaluates `true`)? In other words, I can't figure out what behavior this changes.